### PR TITLE
Remove unused 3rd party dependencies

### PR DESCRIPTION
**Pull request by JLLeitschuh**

### CODE REVIEW

**Comment by JLLeitschuh** [comment]
@bot-gradle test QF

**Review by ljacomet — COMMENTED** [review]
Does any of this clean lead to dependencies that are totally unused?

**Comment by JLLeitschuh** [comment]
@bot-gradle test QF

**Comment by JLLeitschuh** [comment]
@bot-gradle test QF

**Comment by lptr** [comment]
> Does any of this clean lead to dependencies that are totally unused?

I'd also be interested in this.

Also: how did you end up with these exact removals? Is there a tool that did this analysis? If so, can we make it part of our sanity check?

**Review by lptr — APPROVED** [review]

**Comment by melix** [comment]
I like dependency cleanups :) You probably want to remove them from the verification metadata and from the Gradle platform if they are not used anywhere else.

**Comment by JLLeitschuh** [comment]
> Does any of this clean lead to dependencies that are totally unused?

I'm not entirely certain, let me take another pass to check this.

> Also: how did you end up with these exact removals? Is there a tool that did this analysis? If so, can we make it part of our sanity check?

Yes, we introduced the plugin in this PR: #16033

> You probably want to remove them from the verification metadata and from the Gradle platform if they are not used anywhere else.

Will do that too if I can

**Comment by JLLeitschuh** [comment]
>  If so, can we make it part of our sanity check?

Unfortunately, no, it doesn't work correctly with Groovy (yet). https://github.com/autonomousapps/dependency-analysis-android-gradle-plugin/issues/364

**Comment by JLLeitschuh** [comment]
> Does any of this clean lead to dependencies that are totally unused?

I just checked this manually, and it doesn't look this is possible yet.

**Comment by JLLeitschuh** [comment]
@bot-gradle test QF

File: subprojects/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy [hunk 1]
```diff
@@ -37,7 +37,7 @@ import static org.hamcrest.MatcherAssert.assertThat
 
 abstract class DistributionIntegrationSpec extends AbstractIntegrationSpec {
 
-    protected static final THIRD_PARTY_LIB_COUNT = 145
+    protected static final THIRD_PARTY_LIB_COUNT = 144
```
Comment (big-guy) [inline subprojects/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy:40]: which library was removed entirely?

**Comment by JLLeitschuh** [comment]
For those interested. Here is the diff in dependencies after this change.

https://ge.gradle.org/c/h4q4v6izmhnwi/2h6i6zkuitudc/dependencies

In particular, `org.ow2.asm:asm-util:7.3.1` was dropped as a dependency after this change.

**Pull request merged**